modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -466,7 +466,7 @@ conf = {{{packageName}}}.Configuration(
             auth['{{name}}'] = {
                 'type': 'api_key',
                 'in': {{#isKeyInCookie}}'cookie'{{/isKeyInCookie}}{{#isKeyInHeader}}'header'{{/isKeyInHeader}}{{#isKeyInQuery}}'query'{{/isKeyInQuery}},
-                'key': '{{keyParamName}}',
+                'key': '{{#vendorExtensions.x-auth-id-alias}}{{.}}{{/vendorExtensions.x-auth-id-alias}}{{^vendorExtensions.x-auth-id-alias}}{{name}}{{/vendorExtensions.x-auth-id-alias}}',
```

Comment:
Why aren't we using {{keyParamName}} if there is no vendor extension set?

Reply:
> using `keyParamName` is wrong since there can be a collision when using API key in header and query argument under same name: `curl -H 'my-key: header-key' https://example.com/endpoint` vs. `curl https://example.com/endpoint?my-key=query-key`

https://github.com/OpenAPITools/openapi-generator/pull/4940#discussion_r381172053